Classes/Neos/Neos/Ui/Fusion/Helper/NodeInfoHelper.php

```diff
@@ -74,7 +74,13 @@ public function renderNode(NodeInterface $node, ControllerContext $controllerCon
             $nodeInfo['uri'] = $this->uri($node, $controllerContext);
         }
 
-        foreach ($node->getChildNodes() as $childNode) {
+        // child nodes for document tree, respecting the `baseNodeType` filter
+        $documentChildNodes = $node->getChildNodes($this->baseNodeType);
+        // child nodes for content tree, must not include those nodes filtered out by `baseNodeType`
+        $contentChildNodes = $node->getChildNodes('!Neos.Neos:Document');
```

Comment:
This breaks encapsulation, since we introduced the concept of nodeTypeRoles to address NodeTypes like this.

Have a look at https://github.com/neos/neos-ui/blob/ae609f49f4d8c9ae64db3e5b68446aed32878879/Configuration/Settings.yaml#L42

It could work to just replace the explicit `Neos.Neos:Document` with the value in `nodeTypeRole.document`, if we agree that the definition of content in any case is _not being a document_.

Otherwise the filter would have to include both, `nodeTypeRoles.content` and `nodeTypeRoles.contentCollection`